### PR TITLE
Rename `GLOFAS_TEST_RUN` and `test` environment variables

### LINKED ISSUE #94: Clarify `test` and `GMAS_TEST_RUN` variables
For simplicity's sake, can we consolidate a "testing" toggle into a single variable? Looking at the documentation [here](https://github.com/OCHA-DAP/hdx-signals/blob/main/src/alerts/generate_signals.R#L45-L52), it's a little confusing to me how these two interact. For example, if [this `test` flag](https://github.com/OCHA-DAP/hdx-signals/blob/52f083ec5eaa026ce1a8664cd812006be6604ebb/.github/workflows/monitor_conflict.yaml#L13-L19) is `TRUE` on a run, what exactly should happen?

### PR DESCRIPTION
**Pull request by hannahker**
Closes #94. These commits were previously in #131, but want to merge this sooner since it's causing frequent conflicts as changes are made to `main`.

### CODE REVIEW

File: src/email/mailchimp/custom_segmentation.R [hunk 1]
```diff
@@ -31,18 +31,18 @@ box::use(../../utils/get_env[get_env])
 #'
 #' @param indicator_id Unique indicator ID
 #' @param iso3 Vector of ISO3 codes to generate conditions for
-#' @param test Whether or not this is a test email. If so, only sends to contacts
+#' @param dry_run Whether or not this is a dry run email. If so, only sends to contacts
 #'     with the `hdx-signals-test` tag in Mailchimp.
```
Comment (caldwellst) [inline src/email/mailchimp/custom_segmentation.R:35]: Should also change this tag in Mailchimp?

File: src/utils/hs_local.R [hunk 2]
```diff
@@ -1,11 +1,11 @@
 #' Checks if this run is a test run
 #'
-#' `gmas_test_run()` looks for the environment variable `GMAS_TEST_RUN`. This is
+#' `hs_local()` looks for the environment variable `HS_LOCAL`. This is
 #' used in other functions so emails are not sent and data not saved back to the
 #' Google Cloud Storage bucket so test runs do not have any impact on the stored
```
Comment (caldwellst) [inline src/utils/hs_local.R:5]: Edit text here to also reflect it's not about test runs (also not about Google Cloud Storage anymore ha), but just that all changes and outputs are local.

**Review by caldwellst — APPROVED** [review]
All looks good to me, although hard to know if any have been potentially missed...Will probably find out as we go along. I created a new MailChimp tag `hdx-signals-dry-run` that is tagged to yourself and me for now, can also add @zackarno in if he wants to be pinged in all dry runs! We can then change that `hdx-signals-test` to instead reflect `dry-run` and be more clearly aligned with the env variable.

File: src/utils/hs_local.R (hunk 2 above)
Comment (hannahker) [reply src/utils/hs_local.R]: yup good point

File: src/email/mailchimp/custom_segmentation.R (hunk 1 above)
Comment (hannahker) [reply src/email/mailchimp/custom_segmentation.R:35]: Yeah good point! How would the mechanics of that work?

**Pull request merged**